python/datasources/acs_condition.py

```diff
@@ -0,0 +1,509 @@
+
+    def get_filename_race(self, measure, race, is_county):
+        geo = 'COUNTY' if is_county else 'STATE'
+        race = race.replace(" ", "_").upper()
+        return f'{measure.upper()}_BY_RACE_{geo}_{race}.json'
+
+    def get_filename_sex(self, measure, is_county):
+        geo = 'COUNTY' if is_county else 'STATE'
+        return f'{measure.upper()}_BY_SEX_{geo}.json'
+
+    @staticmethod
+    def get_id():
+        return 'ACS_CONDITION'
+
+    @staticmethod
+    def get_table_name():
+        return 'acs_condition'
+
+    # Uploads the ACS data to GCS by providing
+    # the ACS Base URL
+    # Acs Query Params
+    # Standardized Filename
+    #
+    # An example file created in GCS:
+    # HEALTH_INSURANCE_BY_RACE_COUNTY_WHITE_ALONE.json
+    #
+    # Returns:
+    # FileDiff = If the data has changed by diffing the old run vs the new run.
+    def upload_to_gcs(self, bucket, **attrs):
+        # Iterates over the different race ACS variables,
+        # retrieves the race from the metadata merged dict
+        # writes the data to the GCS bucket and sees if file diff is changed
+        file_diff = False
+        for measure, acs_item in ACS_ITEMS.items():
+            for prefix, race in acs_item.prefix_map.items():
+                for county_level in [True, False]:
+                    params = get_all_params_for_group(prefix, county_level)
+                    file_diff = (
+                        url_file_to_gcs.url_file_to_gcs(
+                            self.base_url,
+                            params,
+                            bucket,
+                            self.get_filename_race(measure, race, county_level),
+                        )
+                        or file_diff
+                    )
+
+            for county_level in [True, False]:
+                params = get_all_params_for_group(acs_item.sex_age_prefix, county_level)
+                file_diff = (
+                    url_file_to_gcs.url_file_to_gcs(
+                        self.base_url, params, bucket, self.get_filename_sex(measure, county_level)
+                    )
+                    or file_diff
+                )
+
+        return file_diff
+
+    def write_to_bq(self, dataset, gcs_bucket, **attrs):
+        metadata = census.fetch_acs_metadata(self.base_url)
+        dfs = {}
+        for geo in [NATIONAL_LEVEL, STATE_LEVEL, COUNTY_LEVEL]:
+            for demo in [RACE, AGE, SEX]:
+                table_name = f'by_{demo}_{geo}_processed'
+                df = self.get_raw_data(demo, geo, metadata, gcs_bucket=gcs_bucket)
+                df = self.post_process(df, demo, geo)
+
+                if demo == RACE:
+                    add_race_columns_from_category_id(df)
+
+                dfs[table_name] = df
+
+        for table_name, df in dfs.items():
+            float_cols = [std_col.UNINSURED_PER_100K_COL,
+                          std_col.UNINSURED_POPULATION_PCT,
+                          std_col.UNINSURED_PCT_SHARE_COL]
```

Comment:
should this list include POVERTY metric col names as well?

Reply:
Good catch, kind of weird this didnt cause an issue

Reply:
I think it infers the type of it's not provided. We may not have to explicitly define the float cols actually, because nulls are allowed in bq floats. I wonder if it would auto-cast to ints if it's all whole numbers... hm

Reply:
Actually the fn should have made them strings on bq